ResourceStatusSystem/PolicySystem/Actions/EmailAction.py

```diff
@@ -46,86 +56,57 @@ def run( self ):
-    body += '\n\n'
-    body += '*' * 80
-    body += '\nORGINAL PARAMETERS\n\n'
-    body += '\n'.join( [ '%s : "%s"' % ( key, value ) for key, value in self.decisionParams.items() ] )
-    body += '\n\n'
-    body += '*' * 80
-    body += '\nPOLICIES RUN\n\n'
+    actionResult = self._addtoJSON(self.cacheFile, siteName, dict)
 
-    for policy in self.singlePolicyResults:
+    #returns S_OK() if the record was added successfully using addtoJSON
+    return actionResult
 
-      body += '\n'.join( [ '%s : "%s"' % ( key, value ) for key, value in policy.items() if not key == 'Policy' ] )
-      body += '\n'
-      body += '\n'.join( [ '%s : "%s"' % ( key, value ) for key, value in policy[ 'Policy' ].items() ] )
-      body += '\n\n'
 
-    return self._sendMail( subject, body )
-
-  def _getUserEmails( self ):
+  def _addtoJSON(self, cache_file, siteName, record):
+    ''' Adds a record of a banned element to a local JSON file grouped by site name.
+    '''
 
-    policyActions = RssConfiguration.getPolicyActions()
-    if not policyActions[ 'OK' ]:
-      return policyActions
     try:
-      notificationGroups = policyActions[ 'Value' ][ self.actionName ][ 'notificationGroups' ]
-    except KeyError:
-      return S_ERROR( '%s/notificationGroups not found' % self.actionName )
-
-    notifications = RssConfiguration.getNotifications()
-    if not notifications[ 'OK' ]:
-      return notifications
-    notifications = notifications[ 'Value' ]
-
-    userEmails = []
 
-    for notificationGroupName in notificationGroups:
-      try:
-        userEmails.extend( notifications[ notificationGroupName ][ 'users' ] )
-      except KeyError:
-        self.log.error( '%s not present' % notificationGroupName )
+      if (not os.path.isfile(cache_file)) or (os.stat(cache_file).st_size == 0):
+        #if the file is empty or it does not exist create it and write the first element of the group
+        with open(cache_file, 'w') as f:
+          json.dump({ siteName: [record] }, f)
 
-    return S_OK( userEmails )
+      else:
+        #otherwise load the file
+        with open(cache_file) as f:
+          new_dict = json.load(f)
 
-  def _sendMail( self, subject, body ):
+        #if the site's name is in there just append the group
+        if siteName in new_dict:
+          new_dict[siteName].append(record)
+        else:
+          #if it is not there, create a new group
+          new_dict.update( { siteName: [record] } )
 
-    userEmails = self._getUserEmails()
-    if not userEmails[ 'OK' ]:
-      return userEmails
+        #write the file again with the modified contents
+        with open(cache_file, 'w') as f:
+          json.dump(new_dict, f)
 
-    # User email address used to send the emails from.
-    fromAddress = RssConfiguration.RssConfiguration().getConfigFromAddress()
+      return S_OK()
 
-    for user in userEmails[ 'Value' ]:
+    except ValueError:
+      return S_ERROR("Could not add site to cache file, " + ValueError)
```

Comment:
except ValueError as e:
  return S_ERROR ("blabla: %s"%repr(e))

also you could consider using the new S_ERROR, which takes an error number and a technical message